app/views/invoices/show.html.erb

```diff
@@ -36,4 +36,7 @@
                     turbo_method: :delete,
                     turbo_confirm: "Are you sure?"
                   } %></li>
-</ul>
\ No newline at end of file
+</ul>
+<h2>Share your inovoice</h2>
+
+<li><%= link_to "Share", sharetoken_path(@invoice.token) %></li>
```

Comment:
I'd add some JS here to copy the Share URL, rather than loading it.

Reply:
Have added a function that copies the share URL shows an alert.

Reply:
This function copies the current URL, not the share one.

Here's my script for how I did it, as a start: https://github.com/MattSNZ/invoicer/blob/98881d68c9f4926789d3d62281a848134a10b983/app/views/invoices/show.html.erb#L11-L19